src/d3d12/d3d12_command_list.cpp

```diff
@@ -262,16 +321,31 @@ namespace wr::d3d12
 
 	void Draw(CommandList* cmd_list, unsigned int vertex_count, unsigned int inst_count, unsigned int vertex_start)
 	{
+		for (int i = 0; i < D3D12_DESCRIPTOR_HEAP_TYPE_NUM_TYPES; ++i)
+		{
+			cmd_list->m_dynamic_descriptor_heaps[i]->CommitStagedDescriptorsForDraw(*cmd_list);
```

Comment:
We should consider moving this outside of the low level api. The descriptor heap manager is much more high level and doesn't fit at all with the current implementation of the low level api.

Reply:
Sadly this is a downside of our simple stupid approach since the low level isn't the platform independent part.

Reply:
Either the descriptor heap allocator and stuff "should" be written in the style of the API so its consistent or not part of the low level api at all and be managed by the render system.

Reply:
The question of whether going for a C style low level api was a good idea is debatable. But its to late to change that now.

Reply:
The DynamicDescriptorHeap class is almost made to live inside a command list...moving that out to be inside the render system requires a complete refactor and probably will take way more time than we have. I don't think that's a good idea.

Reply:
I'll put it in the back log.